serverless.yml

```diff
@@ -0,0 +1,133 @@
+service: my-store-app
+
+frameworkVersion: "3"
+
+provider:
+  name: aws
+  runtime: nodejs14.x
+  region: eu-central-1
+  # setup profile for AWS CLI.
+  # profile: node-aws
+
+plugins:
+  - serverless-finch
+  # 'serverless-single-page-app-plugin' is a custom plugin that located .serverless_plugins folder.
+  # Existing plugin (https://www.npmjs.com/package/serverless-single-page-app-plugin) doesn't have invalidate cache feature that often used during CI/CD events.
+  # How to build your own plugins: https://www.serverless.com/framework/docs/providers/aws/guide/plugins#service-local-plugin
+  - serverless-single-page-app-plugin
+  - serverless-s3-cleaner
```

Comment:
As I understand plugins are loaded in the order they defined. 
![image](https://user-images.githubusercontent.com/17138621/191038446-0bc1f1ee-b8db-4ae4-a4aa-03965ec75471.png)

`serverless-s3-cleaner` clears the s3 bucket and defined last in the list. It makes me think that S3 bucket is populated by `serverless-finch` and cleared by `serverless-s3-cleaner` in the end. Is it correct? How does it work in this case?

Reply:
Thank you! Nice to see 5/5.
- Considering .husky/empties - they are mentioned in husky's  config, could not use vscode's conventional commits plugin without them. Could not prevent husky to pick them at all otherwise, despite I'm agree it's not necessary to use vscode and/or its plugin as the 'cli's `git commit` works smooth without `./husky/*` empty files' modes change.
- Considering plugins load order - I'm well sure the `s3-cleaner` plugin is executed in its' own case, e. g., 'npm client remove` and a 'finch' plugin is executed in `deploy` use case, e. g., to put the `dist/` contents to a s3 bucket. Documented is the order of load - e. g., the functions are being loaded, which means functions compiled to me, not called. It may require this stuff to be documented due to the probable clashes of functions and plugins names, e. g., the same plugin name may happen to be used for different files' names in `serverless_plugins/`. Also, some init-purpose code is needed to be executed on a plugin load, which may change the state/configuration of the app, so the same resource may need to be changed by 2+ plugins. But what I'm sure is: this load order does not define the plugins subroutines' execution order on 'npm' targets being performed.

What I'm unsure is - if for this project (and a study course) purposes we have or are allowed to change and submit for PR the:
- package.json?
- package-lock.json?
Or we need to document explicitly`npx sls install` in commit and/or PR messages?

Reply:
- I'd put .husky to `.gitignore` to avoid commiting such files. I beleive that it should not break vscode's conventional commits plugin.
- Thanks for explanation. Seems that I didn't fully understand how plugins work. I will dig deeper :)
- Regarding the `package.json` and `package-lock.json`. I think that it is not forbidden to change them. I beleive that you can even install any other dependencies if you want and it doesn't conflict with task goals.